xbmc/addons/AddonDatabase.cpp

```diff
@@ -324,7 +328,7 @@ bool CAddonDatabase::GetAddons(VECADDONS& addons, const ADDON::TYPE &type /* = A
     if (NULL == m_pDB.get()) return false;
     if (NULL == m_pDS2.get()) return false;
 
-    std::string sql = PrepareSQL("select distinct addonID from addon");
+    std::string sql = PrepareSQL("SELECT DISTINCT a.addonID FROM addon a, addonlinkrepo b WHERE b.idRepo > 0 AND a.id = b.idAddon");
```

Comment:
personally I find this easier to read that a join, but I didn't want to be changing anything in here really, just fix the bug that I saw, so feel free to change it ;)